index.js

```diff
@@ -1,71 +1,109 @@
+import Parser from './src/epubParsers'
+import ResHandler from './src/responseHandlers'
 
 const fileNameRegex = /[0-9]+[.]{1}epub[.]{1}(?:no|)images/
 
-const lambdaEnvVarsClient = new LambdaEnvVars()
+var records
 
-var record, fileName, dateUpdated, putParams, handleResp, records, headParams, kinesis
+exports.handler = async (event, context, callback) => {
+  records = event['Records']
+  let resp
+  if (!records || records.length < 1) {
+    resp = {
+      'status': 500,
+      'code': 'missing_records',
+      'message': 'No records found in event'
+    }
+    ResHandler.resultHandler(resp)
+    return callback(new Error('Kinesis stream failed or contained no records'))
+  } else {
+    await exports.parseRecords(records)
+    return callback(null, 'Successfully parsed records')
+  }
+}
 
-exports.handler = (event, context, callback) => {
-    records = event['Records']
-    for(var i = 0; i < records.length; i++){
-        record = records[i]
-        let payload = JSON.parse(new Buffer.from(record.kinesis.data, 'base64').toString('ascii'))
-        let url = payload['url']
-        let fileNameMatch = fileNameRegex.exec(url)
-        if (!fileNameMatch){
-            handleResp = {
-                "status": 500,
-                "code": "Regex Failure",
-                "message": "Failed to extract file from url " + url
-            }
-            resultHandler(handleResp)
-            continue
-        }
-        let fileName = fileNameMatch[0]
-        let itemID = payload['id']
-        let updated = new Date(payload['updated'])
-        checkForExisting(fileName, updated).then((status) => {
-            axios({
-                method: 'get',
-                url: url,
-                responseType: 'stream'
-            })
-            .then((response) => {
-                epubExplode(fileName, itemID, updated, response)
-                getBuffer(response.data).then((buffer) => {
-                    epubStore(fileName, itemID, updated, 'archive', buffer)
-                })
-                .catch((error) => {
-                    handleResp = {
-                        "status": 500,
-                        "code": "Stream-to-Buffer Error",
-                        "message": error
-                    }
-                    resultHandler(handleResp)
-                })
+exports.parseRecords = async (records) => {
+  let results = records.map(exports.parseRecord)
+  return new Promise((resolve, reject) => {
+    Promise.all(results).then((responses) => {
+      responses.forEach((resp) => {
+        ResHandler.resultHandler(resp)
+      })
+      resolve()
+    })
+  })
+}
+
+exports.runAccessCheck = async (zipData, itemID) => {
+  try{
+    let accessReport = await AccessibilityChecker.getAccessibilityReport(zipData)
+    accessReport['id'] = itemID
+    return {
+      "status": 200,
+      "code": "accessibility",
+      "message": "Created Accessibility Score",
+      "data": accessReport
+    }
+  } catch(err) {
```

Comment:
I wonder if it's cleaner just to rethrow the error here? And similarly in the filename match below and other places, there are a bunch of spots where a 500 error might be returned, would it be simpler to throw errors and have them caught, logged, and converted to a return code in one place?

Reply:
After doing the refactoring suggested below, this is essentially what it does now. It throws Lambda Errors, which are handled/logged at a higher level